### PR TITLE
feat: add communities basic endpoints

### PR DESCRIPTION
**Pull request by AndrePzDn**
# Description

Creation of basic endpoints for communities.

## Type of change

- [x] Create new Community
- [X] Find all communities
- [X] Find one comminity
- [X] Update one community
- [X] Delete one community

# How Has This Been Tested?

To test this feature you can use communities endpoints

- `GET` - `{{baseUrl}}/communities/`
- `GET` - `{{baseUrl}}/communities/{{communityId}}`
- `POST` - `{{baseUrl}}/communities/`
- `PUT` - `{{baseUrl}}/communities/{{communityId}}`
- `DELETE` - `{{baseUrl}}/communities/{{communityId}}`

You can use this template as body in `POST` and `PUT` ones:

```json
{
    "name": "Community",
    "description": "Some description",
    "type": "Yoga",
    "cost": 10,
    "imageUrl": "someurl.com",
    "trainerId": "879451987354198"
}
```

# Links

- [ClickUp Task](https://app.clickup.com/t/868aw3wq7)

### CODE REVIEW

File: apps/engage-hub-service/src/communities/dto/update-community.dto.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    message: `specialization must be one of: ${SpecializationTypes.getAllSpecializations().join(', ')}`,
+  })
+  @IsOptional()
+  type: string;
+
+  @IsNumber()
+  @IsOptional()
+  cost: number;
+
+  @IsString()
```
Comment (JheremyTancara) [inline apps/engage-hub-service/src/communities/dto/update-community.dto.ts]: Redundant imageUrl validation:
@IsString() is not needed when you are already using @IsUrl().

Corrected code example:
@IsUrl()
@IsOptional()
imageUrl: string;

File: apps/engage-hub-service/src/communities/dto/update-community.dto.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { IsEnum, IsOptional, IsNumber, IsString, IsUrl } from 'class-validator';
+import { SpecializationTypes } from 'src/trainer-request/entities/especialization-types.entity';
+
+export class UpdateCommunityDto {
+  @IsString()
+  @IsOptional()
+  name: string;
```
Comment (JheremyTancara) [inline apps/engage-hub-service/src/communities/dto/update-community.dto.ts]: It is recommended to use ? and @IsOptional() together on optional properties of a DTO: ? tells TypeScript that they are not mandatory, and @IsOptional() prevents errors in class-validator if they are absent. 

Example:
@IsString()
@IsOptional()
name?: string;

File: apps/engage-hub-service/src/communities/dto/update-community.dto.ts (hunk 1 above)
Comment (AndrePzDn) [reply apps/engage-hub-service/src/communities/dto/update-community.dto.ts]: It has been fixed in new commit, please review it.

File: apps/engage-hub-service/src/communities/dto/update-community.dto.ts (hunk 2 above)
Comment (AndrePzDn) [reply apps/engage-hub-service/src/communities/dto/update-community.dto.ts]: It has been fixed in new commit, please review it.

**Review by GastonGutierrezC — APPROVED** [review]

**Review by JheremyTancara — APPROVED** [review]

**Pull request merged**